var/spack/repos/builtin/packages/py-nodeenv/package.py

```diff
@@ -0,0 +1,18 @@
+# Copyright 2013-2019 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class PyNodeenv(PythonPackage):
+    """Node.js virtual environment"""
+
+    homepage = "https://github.com/ekalinin/nodeenv"
+    url      = "https://pypi.io/packages/source/n/nodeenv/nodeenv-1.3.3.tar.gz"
+
+    version('1.3.3', sha256='ad8259494cf1c9034539f6cced78a1da4840a4b157e23640bc4a0c0546b0cb7a')
+
+    depends_on('python@2.6:2.8,3.3:', type=('build', 'run'))
```

Comment:
I would actually remove this, there is no `python_requires` in the `setup.py` that enforces this.

Reply:
```
        'Programming Language :: Python',
        'Programming Language :: Python :: 2',
        'Programming Language :: Python :: 2.6',
        'Programming Language :: Python :: 2.7',
        'Programming Language :: Python :: 3',
        'Programming Language :: Python :: 3.3',
        'Programming Language :: Python :: 3.4',
        'Programming Language :: Python :: 3.5',
        'Programming Language :: Python :: 3.6',
```
are these also just advisory? They seem to be up to a point, as it works for me with py3.7 but I figure that 3.0-3.2 were omitted intentionally

Reply:
The Python versions listed here are what the developer claims are supported. They are not enforced in any way, so you can actually try to build the package with Python 2.5 if you want to, although it's not guaranteed to work. If the developer uses `python_requires` (supported in newer versions of setuptools), that is actually enforced, and the build will fail letting you know that this version of Python is not supported.

My personal philosophy has been to only use `python_requires` in the `depends_on`, since as you've noticed, Python 3.7+ work just fine. I would be open to your philosophy as well, since not everyone uses `python_requires` yet. This is even more important for packages that likely only support Python 2.

Reply:
I'll stick to your solution, if something breaks we will see it and it should be (relatively) easy to fix